File: solutions/csharp/space-age/1/SpaceAge.cs
using System.Collections.ObjectModel;

public class SpaceAge
{
    private const int EARTH_SECONDS_IN_YEAR = 31_557_600;
    private readonly ReadOnlyDictionary<string, double> orbital_ratios = new ReadOnlyDictionary<string, double>(new Dictionary<string, double>()
    {
        { "Mercury", 0.2408467 },
        { "Venus",   0.61519726 },
        { "Mars",   1.8808158 },
        { "Jupiter", 11.862615 },
        { "Saturn",  29.447498 },
        { "Uranus",  84.016846 },
        { "Neptune", 164.79132 }
    });
    private readonly int seconds;

    public SpaceAge(int seconds)
    {
        this.seconds = seconds;
    }

    public double OnEarth()
        => seconds / (double)EARTH_SECONDS_IN_YEAR;


    public double OnMercury()
        => OnEarth() / orbital_ratios["Mercury"];

    public double OnVenus()
        => OnEarth() / orbital_ratios["Venus"];

    public double OnMars()
        => OnEarth() / orbital_ratios["Mars"];

    public double OnJupiter()
        => OnEarth() / orbital_ratios["Jupiter"];

    public double OnSaturn()
        => OnEarth() / orbital_ratios["Saturn"];

    public double OnUranus()
        => OnEarth() / orbital_ratios["Uranus"];

    public double OnNeptune()
        => OnEarth() / orbital_ratios["Neptune"];
}

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds){
  const int year_in_seconds_earth = 31557600;
  float result = 0.0;
  float year_in_seconds = 0.0;
  switch(planet){
    case MERCURY:
      year_in_seconds = 0.2408467 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case VENUS:
      year_in_seconds = 0.61519726 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case MARS:
      year_in_seconds = 1.8808158 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case JUPITER:
      year_in_seconds = 11.862615 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case SATURN:
      year_in_seconds = 29.447498 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case URANUS:
      year_in_seconds = 84.016846 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case NEPTUNE: 
      year_in_seconds = 164.79132 * year_in_seconds_earth;
      result = (float)seconds / year_in_seconds;
      break;
    case EARTH:
      result = (float)seconds / year_in_seconds_earth;
      break;
    default:
      return -1.0;
  }
  return result;
}

